cli/internal/run/run.go

```diff
@@ -806,12 +806,10 @@ func replayLogs(logger hclog.Logger, prefixUi cli.Ui, runOptions *RunOptions, lo
 	}
 	defer f.Close()
 	if outputLogsMode != NoLogs {
-		scan := bufio.NewScanner(f)
 		if outputLogsMode == HashLogs {
-			//Writing to Stdout only the "cache hit, replaying output" line
-			scan.Scan()
-			prefixUi.Output(ui.StripAnsi(string(scan.Bytes())))
+			prefixUi.Output(fmt.Sprintf("%scache hit, suppressing output %s", logPrefix, ui.Dim(hash)))
```

Comment:
This line seems a little weird: we're outputting from something called `prefixUi`, but we're also manually adding a prefix. 

I think we might want to refactor this a little to be less confusing. I think what's going on is the following:
 * the calling code has two `Ui` instances, one prefixed, one not
 * we're not actually passing in the prefixed one, because we expect the prefix to be in the logs already
  * we're then manually adding the prefix in the case where we are synthesizing a log line

Here's my proposal: 
 * check the `outputLogsMode` before calling `replayLogs`. 
 * If we're in 'NoLogs' mode, great, we're done
 * create a separate function to just output the cache hit for `HashLogs` mode. We're already not using the file contents in that case, so don't even open the file. Pass in the correct `Ui` instance that already has the prefix baked in (`targetUi`, I think). It doesn't even need to be in a goroutine, we could just call it directly. In fact, it might not even need to be function, it might even be just a single line
  * If we're in `FullLogs` mode, use the existing `replayLogs`
  * Remove the case for `NoLogs` in `replayLogs`

Reply:
sgtm

Reply:
That refactor really cleans it up. Thanks for the suggestion!